commit create book repository

krcg93 opened the pull request:
## Summary
- Completed book repository

## Trello Card
https://trello.com/c/jzCfi1qh/15-crear-un-repositorio-de-books

krcg93 committed:
commit create book repository

juandbc commented on src/main/java/wolox/training/repositories/BookRepository.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.data.repository.query.Param;
+import wolox.training.models.Book;
+
+public interface BookRepository extends JpaRepository<Book, Integer> {
```
We should use the `Long` data type for the id attribute
```suggestion
public interface BookRepository extends JpaRepository<Book, Long> {
```

juandbc commented on src/main/java/wolox/training/repositories/BookRepository.java:
```diff
@@ -0,0 +1,12 @@
+
+public interface BookRepository extends JpaRepository<Book, Integer> {
+
+    @Query("SELECT b FROM Book b WHERE LOWER(b.author) = LOWER(:author)")
+    Book bookByAuthor(@Param("author") String author);
```
Two things here:

1. It's better return the result wrapped in an `Optional`. Thus, we can avoid null checks
2. `@Query` is fine, but we should use for complexies queries or when [Query methods](https://docs.spring.io/spring-data/data-jpa/docs/current/reference/html/#repositories.query-methods) doesn't support the keyword one wants to use.
Change to use query methods.

juandbc reviewed (changes requested):
Last, we at Wolox try to use commit messages in english (this depends of the project). Follow this [write good commits](https://chris.beams.io/posts/git-commit/) guide to improve your commit's messages

krcg93 committed:
change is from custom query to automatic query, and change data type of the id attribute
